engineio/asyncio_server.py

```diff
@@ -63,6 +67,13 @@ def attach(self, app, engineio_path='engine.io'):
         engineio_path = engineio_path.strip('/')
         self._async['create_route'](app, self, '/{}/'.format(engineio_path))
 
+    async def shutdown(self):
+        """Shutdown the Engine.IO server."""
+        self.running = False
+
+        self.__service_task.cancel()
+        await self.__service_task
```

Comment:
How long does this take? It would seem you could be waiting up to a full minute until the task exits.